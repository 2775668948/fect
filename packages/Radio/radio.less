.fay-radio {
  display: flex;
  align-items: flex-start;
  position: relative;
  width: initial;
  input {
    opacity: 0;
    overflow: hidden;
    visibility: hidden;
    width: 1px;
    height: 1px;
    top: -1000px;
    right: -1000px;
    position: fixed;
  }
  label {
    cursor: pointer;
    color: var(--accents-2);
    &.disabled {
      color: var(--accents-5);
      cursor: not-allowed;
    }
  }
  .fay-radio-name {
    font-size: var(--radioSize);
    font-weight: bold;
    user-select: none;
    display: inline-flex;
    align-items: center;
  }
  .fay-radio-point {
    height: var(--radioSize);
    width: var(--radioSize);
    border: 1px solid var(--accents-7);
    border-radius: 50%;
    transition: all 0.2s ease 0s;
    position: relative;
    display: inline-block;
    transform: scale(0.875);
    margin-right: calc(var(--radioSize) * 0.375);

    &::before {
      background-color: var(--accents-2);
      content: '';
      position: absolute;
      transform: scale(0);
      border-radius: 50%;
      height: var(--radioSize);
      width: var(--radioSize);
    }
    &.disabled::before {
      background-color: var(--accents-5);
    }
    &.active::before {
      transform: scale(0.875);
      transition: all 0.2s ease 0s;
    }
  }
}
