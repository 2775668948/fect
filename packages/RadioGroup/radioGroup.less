.fay-radio-group {
  display: flex;
  flex-direction: column;
  &.useRow {
    flex-direction: row;
  }
}
