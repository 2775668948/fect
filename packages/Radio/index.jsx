import { computed, onMounted, ref, toRefs, watch, watchEffect } from 'vue'
import { createNameSpace, theme, validator,useProvider } from '../utils'

const { normalSizes } = theme

const [createComponent] = createNameSpace('Radio')

const READNONLY_RADIO_GROUP_KEY = 'radioGroupKey'

import './radio.less'


const queryRadioSize = (radioSize)=>{
  const size = {
    mini:'12px',
    small:'14px',
    medium:'16px',
    large:'18px',
  }
  return size[radioSize]
}


export default createComponent({
  props: {
    checked: Boolean,
    disabled:Boolean,
    value: { type: String ,required:true },
    size: {
      type: String,
      validator: validator.enums(normalSizes),
      default: 'medium',
    },
  },
  emtis: ['change'],
  setup(props, { attrs, slots, emit }) {
    
    const { checked, value, size ,disabled } = toRefs(props)
    const { ctx } = useProvider(READNONLY_RADIO_GROUP_KEY)
    const radioValue = ref(value.value)
    const radioSize = ref(size.value)
    const isDisabled = ref(disabled.value)
    const selfChecked = ref(!!checked.value)


    const changeStatus = ()=>{
      isDisabled.value = ctx.props.disabled
      radioSize.value = ctx.props.size
      radioValue.value = ctx.props?.initialValue || null
      selfChecked.value = (radioValue.value === value.value)
    }

    if (ctx){
      watchEffect(()=>{
        changeStatus()
      })
    }
   
    const handlerChange = (e) => {
      if (isDisabled.value) return
      const radioEvent = {
        target:{
          checked:!selfChecked.value,
        },
        stopPropagation: e.stopPropagation,
        preventDefault: e.preventDefault,
        nativeEvent: e,
      }
      selfChecked.value = !selfChecked.value;
      if (ctx){
        console.log(value.value)
        ctx.updateState && ctx.updateState(radioValue.value)
      }
      emit('change', radioEvent)
    }

    const calcRadioSize = computed(()=>{
      const _size = queryRadioSize(radioSize.value)
      const style = {}
      style['--radioSize'] = _size
      return style
    })

    return () => (
      <div className={`fay-radio ${attrs?.class ? attrs.class : ''}`} style={calcRadioSize.value}>
        <label className={` ${isDisabled.value ? 'disabled' : ''}`}>
          <input
            {...attrs}
            type="radio"
            value={radioValue.value}
            checked={selfChecked.value}
            onChange={handlerChange}
            disabled={isDisabled.value}
          ></input>
          <span className={'fay-radio-name'}>
            <span 
              className={`fay-radio-point ${isDisabled.value ? 'disabled' : ''} ${selfChecked.value ? 'active' : ''}`}  />
            {slots.default?.()}
          </span>
        
        </label>
      </div>
    )
  },
})
