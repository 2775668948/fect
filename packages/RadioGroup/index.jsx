import { computed, toRefs ,ref } from 'vue'
import { createNameSpace, theme, validator,createProvider } from '../utils'

const { normalSizes } = theme

const [createComponent] = createNameSpace('RadioGroup')

const READNONLY_RADIO_GROUP_KEY = 'radioGroupKey'

import './radioGroup.less'

export default createComponent({
  props: {
    initialValue: [String, Number],
    useRow: Boolean,
    disabled: Boolean,
    size: {
      type: String,
      validator: validator.enums(normalSizes),
      default: 'medium',
    },
  },
  emits: ['change'],
  setup(props, { attrs, slots,emit }) {
    const { initialValue, useRow, disabled, size } = toRefs(props)
    const selfvalueGroup = ref(initialValue?.value)
    const groupSize = ref(size.value)
    const disabledAll = ref(disabled.value)
    const updateState = (nextVal)=>{
      sefvalueGroup.value = nextVal
      emit(nextVal)
    }
    const { provider } = createProvider(READNONLY_RADIO_GROUP_KEY)
    provider({ updateState,disabledAll,groupVal:selfvalueGroup,inGroup:true ,groupSize })
    return () => (
      <>
        <div 
          {...attrs} 
          className={`fay-radio-group 
            ${useRow.value ? 'useRow' : ''}
            ${attrs?.class ? attrs.class : ''}`}>
          {slots.default?.()}
        </div>
        
      </>
    )
  },
})
