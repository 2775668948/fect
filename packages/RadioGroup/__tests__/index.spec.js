import { mount } from '@vue/test-utils'
import RadioGroup from '../index'
import Radio from '../../Radio'

describe('RadioGroup', () => {
  it('should be render as element', () => {
    const radioGroup = mount(RadioGroup)
    expect(radioGroup.html()).toMatchSnapshot()
    expect(() => radioGroup.unmount()).not.toThrow()
  })

  it('should be support initialValue', () => {
    const wrapper = mount(
      <RadioGroup initialValue={1}>
        <Radio value={1}></Radio>
        <Radio value={2}></Radio>
      </RadioGroup>,
    )

    expect(wrapper.html()).toMatchSnapshot()
    const els = wrapper.findAll('.fay-radio-point')
    expect(els[0].classes('active')).toBeTruthy()
  })

  it('should be support useRow', () => {
    const wrapper = mount(<RadioGroup useRow={true}></RadioGroup>)
    expect(wrapper.find('.fay-radio-group').classes('useRow')).toBe(true)
  })

  it('should be support disabled', () => {
    const wrapper = mount(
      <RadioGroup disabled={true}>
        <Radio value={1} />
        <Radio value={2} />
      </RadioGroup>,
    )

    const labels = wrapper.findAll('label')
    expect(labels[0].classes('disabled')).toBeTruthy()
    expect(labels[1].classes('disabled')).toBeTruthy()
  })

  it('should be support differentSizes', () => {
    const wrapper = mount({
      setup() {
        return { sizeList: ['mini', 'small', 'medium', 'large'] }
      },
      render(h) {
        return (
          <>
            {this.sizeList.map((item) => (
              <RadioGroup size={item} key={item}>
                <Radio value={item} />
              </RadioGroup>
            ))}
          </>
        )
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
    expect(() => wrapper.unmount()).not.toThrow()
  })

  it('should be support emit change', () => {
    const wrapper = mount({
      setup() {
        const test = ref(0)
        const handlerChange = (next) => (test.value = next)
        return {
          test,
          handlerChange,
        }
      },
      render() {
        return (
          <>
            <RadioGroup onChange={this.handlerChange}>
              <Radio value={1} />
              <Radio value={2} />
            </RadioGroup>
          </>
        )
      },
    })
    
  })
})
