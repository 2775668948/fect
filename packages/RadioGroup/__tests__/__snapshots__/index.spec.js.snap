// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioGroup should be render as element 1`] = `
<div class="fay-radio-group 
            
            ">
  <!---->
</div>
`;

exports[`RadioGroup should be support differentSizes 1`] = `
<div class="fay-radio-group 
            
            ">
  <div class="fay-radio " style="--radioSize: 12px;"><label class=" "><input type="radio" value=""><span class="fay-radio-name"><span class="fay-radio-point  "></span>
      <!----></span>
    </label></div>
</div>
<div class="fay-radio-group 
            
            ">
  <div class="fay-radio " style="--radioSize: 14px;"><label class=" "><input type="radio" value=""><span class="fay-radio-name"><span class="fay-radio-point  "></span>
      <!----></span>
    </label></div>
</div>
<div class="fay-radio-group 
            
            ">
  <div class="fay-radio " style="--radioSize: 16px;"><label class=" "><input type="radio" value=""><span class="fay-radio-name"><span class="fay-radio-point  "></span>
      <!----></span>
    </label></div>
</div>
<div class="fay-radio-group 
            
            ">
  <div class="fay-radio " style="--radioSize: 18px;"><label class=" "><input type="radio" value=""><span class="fay-radio-name"><span class="fay-radio-point  "></span>
      <!----></span>
    </label></div>
</div>
`;

exports[`RadioGroup should be support initialValue 1`] = `
<div class="fay-radio-group 
            
            ">
  <div class="fay-radio " style="--radioSize: 16px;"><label class=" "><input type="radio" value="1"><span class="fay-radio-name"><span class="fay-radio-point  active"></span>
      <!----></span>
    </label></div>
  <div class="fay-radio " style="--radioSize: 16px;"><label class=" "><input type="radio" value="1"><span class="fay-radio-name"><span class="fay-radio-point  "></span>
      <!----></span>
    </label></div>
</div>
`;
